src/firmware/app/primitives/chick_alignment.c

```diff
@@ -0,0 +1,108 @@
+#include "firmware/app/primitives/chick_alignment.h"
+
+#include "firmware/app/control/bangbang.h"
+#include "firmware/app/control/control.h"
+#include "firmware/app/control/physbot.h"
+#include "firmware/app/primitives/primitive.h"
+#include "firmware/shared/physics.h"
+#include "firmware/shared/util.h"
+
+#define TIME_HORIZON 0.05f  // in seconds
+
+typedef struct ShootAlignmentState
+{
+    float destination[3], major_vec[2], minor_vec[2], total_rot;
+} ShootAlignmentState_t;
+DEFINE_PRIMITIVE_STATE_CREATE_AND_DESTROY_FUNCTIONS(ShootAlignmentState_t)
+
+/**
+ * Scales the major acceleration by the distance from the major axis and the
+ * amount required left to rotate.
+ *
+ * @pre Total rotation and the distance vector should
+ * not be zero so as to avoid divide by zero errors.
+ *
+ * @param pb [in/out] The PhysBot data container that contains information about the
+ * major and minor axis.
+ */
+void scale(PhysBot *pb)
+{
+    float maj_disp        = fabsf(pb->maj.disp) - ROBOT_RADIUS;
+    float distance_vector = sqrtf(powf(maj_disp, 2) + powf(pb->min.disp, 2));
+    if (distance_vector != 0)
+    {
+        float abs_factor        = fabsf(pb->min.disp) / distance_vector;
+        float minor_axis_factor = 1 - abs_factor;
+        pb->maj.accel *= minor_axis_factor;
+    }
+}
+
+/**
+ * Determines the rotation acceleration after setup_bot has been used and
+ * plan_move has been done along the minor axis. The minor time from bangbang
+ * is used to determine the rotation time, and thus the rotation velocity and
+ * acceleration. The rotational acceleration is clamped under the MAX_T_A.
+ *
+ * @param pb [in/out] The PhysBot data container that should have minor axis time and
+ * will store the rotational information
+ * @param avel The rotational velocity of the bot
+ */
+void plan_shoot_rotation(PhysBot *pb, float avel)
+{
+    pb->rot.time = (pb->min.time > TIME_HORIZON) ? pb->min.time : TIME_HORIZON;
+    // 1.6f is a magic constant
+    pb->rot.vel   = 1.6f * pb->rot.disp / pb->rot.time;
+    pb->rot.accel = (pb->rot.vel - avel) / TIME_HORIZON;
+    limit(&pb->rot.accel, MAX_T_A);
+}
+
+void app_chick_alignment_start(void *void_state_ptr, FirmwareWorld_t *world,
+                               float x_destination, float y_destination,
+                               float alignment_angle)
```

Comment:
Similar comment to the move stuff, why not just call this `align_to_ball_primitive` instead? Stacking primitives in this way seems a little more comprehensible, at least from an interface perspective?

Reply:
We could expose it in the same way, with a public `ALIGN_TO_BALL` primitive.

Reply:
ran into this error: 
```
firmware/app/primitives/kick_primitive.c:37:54: error: initializer element is not constant
                                     .create_state  = ALIGN_TO_BALL.create_state,
                                                      ^~~~~~~~~~~~~
firmware/app/primitives/kick_primitive.c:37:54: note: (near initialization for 'KICK_PRIMITIVE.create_state')
```
so only did renaming to align_to_ball